src/gui/kbwidget.cpp

```diff
@@ -281,6 +299,17 @@ void KbWidget::modesList_reordered(){
     currentProfile->modes(newModes);
 }
 
+void KbWidget::batteryTrayBox_stateChanged(int state){
+    QString batterySettingsPath("Devices/%1/batteryTrayIcon");
+    CkbSettings::set(batterySettingsPath.arg(device->usbSerial), state);
```

Comment:
The reason this "doesn't work", is because you're not supposed to access the `Devices` config group directly.

The device class overwrites everything every time it has to save something, and something it doesn't know about gets removed.

In order to store per-device settings, they need to be stored locally in the `Kb` class and then saved/loaded using the `Kb::save()` and `Kb::load()` functions, which are called automatically when the GUI autosaves.
The autosave takes place when a setter function in the class is called that sets _needSave to true.

You should save your setting at the end of the Kb::save function and restore it at the end of the load().
https://github.com/ckb-next/ckb-next/blob/e961dd02df7b97ce59c5c33cb65b8cc6b7219cb6/src/gui/kb.cpp#L311-L326

And finally access it here through the Kb class.